### ISSUE TITLE
New order emails from shop with client as CC

### ISSUE THREAD

**Issue by adam3278** [freescout-help-desk/freescout#4358]
Hello,
I have a small problem. My shop sends new order confirmation emails to main freescout mailbox and the customer. The customer is set as CC and it was possible to reply to they when using Outlook. App was clever enough to detect back email and prevented sending it again. The result was quite good, but after switch to Freescout the system receives worker answers and combines them into the conversation. I cannot delete them and they keep changing the conversation status to Active. The perfect solution would be to send emails from WooCommerce via Freescout or fetch new emails from sent folder. I cannot set the customer as sender due to dkim. Can anyone help me?

**Comment by freescout-help** [comment]
> the system receives worker answers and combines them into the conversation. I cannot delete them and they keep changing the conversation status to Active.

This is not quite clear.

Maybe specifying the customer email in `Reply-To` header when sending the email would help.

**Comment by adam3278** [comment]
Ok, I explain this in moment. Imagine situation as this:
1. Your shop sends an order confirmation email to you and your customer (as BCC).
2. You want to reply to this email, but you're the sender also, so Outlook detect this and prevents you sending to yourself (simply ignores this email). But, your client was your BCC, so they receives the email.
3. Now, when the client replies, they are the sender and you're receiver, so everything's normal since then.

Why this happens:
1. I cannot send the confirmation email as my customer to me (the sender server is not authorized to use this email) from shop. Even if I can, my customer should be mailed from me, not from theirself. I can create two emails, but we're stuck with first problem - I cannot send email from customer address to me.
2. I want to reply to the customer directly from freescout with confirmation email attached.
3. I want to know what's going on in my company, so freescout is central work management platform. When the order arrives to it, we know we have to do something with it - no one is checking woocommerce orders panel.
4. WooCommerce is sending the emails using smtp, because there is no possibility to use Freescout as mailing service
5. Assuming all above, I have to start a new ticket in Freescout every time there is a new order in my shop, but the customer must me set as recipient in Freescout (exactly as it would be send via freescout with status change to Active).

This is a common situation. Most of companies are using emails as new orders data source. It acts as notification.

Thanks in advance for help. You're making awesome job and we love your extensions, even making own ones (subfolders extension for Custom Folders module would be published as open source in a moment).

**Comment by adam3278** [comment]
to simplify it a little more:
1. shop sends email as Mailbox A to Mailbox A and Mailbox B
2. Freescout (Mailbox A) receives the email from Mailbox A
3. Worker replies to this email, but the app should ignore Mailbox A as receiver and send it to Mailbox B only.
4. In the best scenario, the app would remove its address (Mailbox A) from ticket, leaving proper Mailbox B. Then the WooCommerce integration module along with customer profile would work just fine.

**Comment by freescout-help** [comment]
Create `noreply@your-shop.com` address (do not add it to FreeScout as a mailbox) and send order confirmation from this address.

**Comment by adam3278** [comment]
Ok, no problem, but then the no-reply email would be the side of the conversation in Freescout, yes?

**Comment by adam3278** [comment]
It would be enough to have ability to change client from Workflows module. I would filter the emails by sender and trigger automatic action of CC to client change.

**Comment by adam3278** [comment]
I found a solution - https://github.com/aarhus/AarhusChangeCustomer/. This wonderful developer created the extension I was exactly looking for.

Now, there is another problem. If I want to reply to this modified conversation, the original sender is placed in CC addresses. I tried to remove it via this simple code:
```php
    $ccs = array_filter($conversation->getCcArray(), function ($recipient) use ($original_customer_email) {
      return !str_contains($recipient, $original_customer_email);
    });
    $conversation->setCc($ccs);
    $bccs = array_filter($conversation->getBccArray(), function ($recipient) use ($original_customer_email) {
      return !str_contains($recipient, $original_customer_email);
    });
    $conversation->setBcc($bccs);
    $conversation->save();
```
The snippet is triggered by "conversation.created_by_customer" action, after customer change via plugin (for tests I placed it inside it). I tried to find the place where the original customer is placed in CC, but I cannot achieve it. Maybe, I should change the first conversation thread "From" and "To" also? This should be less painful to workers than the need of removing shop email every time.
